src/ert/sensitivity_analysis/design_matrix.py

```diff
@@ -0,0 +1,230 @@
+
+import pandas as pd
+import xarray as xr
+
+from ert.config.gen_kw_config import GenKwConfig, TransformFunctionDefinition
+
+if TYPE_CHECKING:
+    from ert.config import (
+        ErtConfig,
+    )
+    from ert.storage import LocalEnsemble, LocalStorage
+
+DESIGN_MATRIX_GROUP = "DESIGN_MATRIX"
+
+
+def read_design_matrix(
+    ert_config: ErtConfig,
+    xlsfilename: Path | str,
+    designsheetname: str = "DesignSheet01",
+    defaultssheetname: str = "DefaultValues",
+) -> pd.DataFrame:
+    """
+    Reads out all file content from different files and create dataframes
+    """
+    design_matrix_sheet = _read_excel(xlsfilename, designsheetname)
+    if "REAL" in design_matrix_sheet.columns:
+        design_matrix_sheet.set_index(design_matrix_sheet["REAL"])
+        del design_matrix_sheet["REAL"]
+    try:
+        _validate_design_matrix_header(design_matrix_sheet)
+    except ValueError as err:
+        raise ValueError(f"Design matrix not valid, error: {err!s}") from err
+
+    # Todo: Check for invalid realizations, drop them maybe?
+
+    if designsheetname == defaultssheetname:
+        raise ValueError("Design-sheet and defaults-sheet can not be the same")
+
+    # This should probably handle/(fill in) missing values in design_matrix_sheet as well
+    defaults = _read_defaultssheet(xlsfilename, defaultssheetname)
+    for k, v in defaults.items():
+        if k not in design_matrix_sheet.columns:
+            design_matrix_sheet[k] = v
+
+    # ignoring errors here is deprecated in pandas, should find another solution
+    # design_matrix_sheet = design_matrix_sheet.apply(pd.to_numeric, errors="ignore")
+
+    parameter_groups = defaultdict(list)
+    parameter_map = []
+    all_param_configs = [
+        param_group
+        for param_group in ert_config.ensemble_config.parameter_configuration
+        if isinstance(param_group, GenKwConfig)
+    ]
+    errors = {}
+    for param in design_matrix_sheet.columns:
+        par_gp = []
+        for param_group in all_param_configs:
+            if param in param_group:
+                par_gp.append(param_group.name)
+
+        if not par_gp:
+            parameter_name = "DESIGN_MATRIX"
+            parameter_groups[parameter_name].append(param)
+            parameter_map.append((parameter_name, param))
+        elif len(par_gp) == 1:
+            parameter_name = par_gp[0]
+            parameter_groups[parameter_name].append(param)
+            parameter_map.append((parameter_name, param))
+        else:
+            errors[param] = par_gp
+
+    if errors:
+        msg = ""
+        for key, value in errors.items():
+            msg += (
+                f"The following parameter '{key}' was found in multiple"
+                f" GenKw parameters groups: {value}."
+            )
+        raise ValueError(msg)
+    design_matrix_sheet.columns = pd.MultiIndex.from_tuples(parameter_map)
+    return design_matrix_sheet
+
+
+def initialize_parameters(
```

Comment:
Suggest moving this code around a bit, so it is more in line with the `sample_prior` function.